nomos-da/network/core/src/swarm/common/monitor.rs

```diff
@@ -99,9 +99,47 @@ impl From<&SamplingEvent> for MonitorEvent {
     }
 }
 
+pub mod dto {
+    use std::collections::HashMap;
+
+    use fixed::types::U57F7;
+    use libp2p::PeerId;
+    use serde::{Deserialize, Serialize};
+
+    #[derive(Debug, Deserialize, Eq, PartialEq, Serialize)]
+    pub struct MonitorStats(pub HashMap<PeerId, PeerStats>);
+
+    impl From<&HashMap<PeerId, super::PeerStats>> for MonitorStats {
+        fn from(stats: &HashMap<PeerId, super::PeerStats>) -> Self {
+            Self(stats.iter().map(|(k, v)| (*k, v.into())).collect())
+        }
+    }
+
+    #[derive(Copy, Clone, Debug, Deserialize, Eq, PartialEq, Serialize)]
+    pub struct PeerStats {
```

Comment:
Just curious, why couldn't `PeerStats` from monitor be used? With serde we could use  `#[serde(skip)]` for `last_...` fields.

Reply:
Yeah, you're correct, for passing it via channel we don't need it. Thanks for the explanation, this solution is great.